components/browser/toolbar/src/test/java/mozilla/components/browser/toolbar/BrowserToolbarTest.kt

```diff
@@ -229,6 +230,36 @@ class BrowserToolbarTest {
         assertTrue(toolbar.editToolbar.visibility == View.GONE)
     }
 
+    @Test
+    fun `toolbar will switch back to display mode if URL commit listener returns true`() {
+        val toolbar = BrowserToolbar(RuntimeEnvironment.application)
+        toolbar.setOnUrlCommitListener { true }
+        toolbar.editMode()
+
+        assertTrue(toolbar.displayToolbar.visibility == View.GONE)
```

Comment:
nit: You could use `toolbar.editToolbar.isGone()` and `toolbar.editToolbar.isVisible()` extension functions :)

Reply:
Yeah, true. I'm less concerned about the "complicated, verbose way" in tests since I want to test a very specific thing and not call into half our utilities. :)